cvmfs/object_fetcher.h

```diff
@@ -319,7 +336,9 @@ class HttpObjectFetcher :
     object_file->clear();
 
     const std::string url  = BuildUrl(object_hash, hash_suffix);
-    const std::string dest = temporary_directory_ + "/" + object_hash.ToString();
+    const std::string dest = CreateTempPath(temporary_directory_ + "/" +
```

Comment:
Would be more secure to use CreateTempFile and to use the JobInfo constructor working with `FILE *`. That also saves one pair of open/close functions.